app/src/main/java/org/oppia/android/app/player/state/StateFragmentPresenter.kt

```diff
@@ -459,31 +462,58 @@ class StateFragmentPresenter @Inject constructor(
   private fun showHintsAndSolutions(helpIndex: HelpIndex, isCurrentStatePendingState: Boolean) {
     if (!isCurrentStatePendingState) {
       // If current state is not the pending top state, hide the hint bulb.
-      viewModel.setHintOpenedAndUnRevealedVisibility(false)
+      setHintOpenedAndUnRevealed(false)
       viewModel.setHintBulbVisibility(false)
     } else {
       when (helpIndex.indexTypeCase) {
         HelpIndex.IndexTypeCase.NEXT_AVAILABLE_HINT_INDEX -> {
           viewModel.setHintBulbVisibility(true)
-          viewModel.setHintOpenedAndUnRevealedVisibility(true)
+          setHintOpenedAndUnRevealed(true)
         }
         HelpIndex.IndexTypeCase.LATEST_REVEALED_HINT_INDEX -> {
           viewModel.setHintBulbVisibility(true)
-          viewModel.setHintOpenedAndUnRevealedVisibility(false)
+          setHintOpenedAndUnRevealed(false)
         }
         HelpIndex.IndexTypeCase.SHOW_SOLUTION -> {
           viewModel.setHintBulbVisibility(true)
-          viewModel.setHintOpenedAndUnRevealedVisibility(true)
+          setHintOpenedAndUnRevealed(true)
         }
         HelpIndex.IndexTypeCase.EVERYTHING_REVEALED -> {
-          viewModel.setHintOpenedAndUnRevealedVisibility(false)
+          setHintOpenedAndUnRevealed(false)
           viewModel.setHintBulbVisibility(true)
         }
         else -> {
-          viewModel.setHintOpenedAndUnRevealedVisibility(false)
+          setHintOpenedAndUnRevealed(false)
           viewModel.setHintBulbVisibility(false)
         }
       }
     }
   }
+
+  private fun setHintOpenedAndUnRevealed(isHintRevealed: Boolean) {
+    viewModel.setHintOpenedAndUnRevealedVisibility(isHintRevealed)
+    if (isHintRevealed) {
+      val hintBulbAnimation = AnimationUtils.loadAnimation(
+        context,
+        R.anim.hint_bulb_animation
+      )
+      binding.hintBulb.startAnimation(
+        hintBulbAnimation
+      )
+
+      // The hint bulb will blink 5 times in duration of 6 sec, stop for next 24 sec and again
+      // start repeating.
```

Comment:
Should this be repeated every 30 seconds? The current implementation will only repeat it once.

Reply:
Verified the original solution didn't repeat, but the changes do.